web/testing/issue005/header.js

```diff
@@ -0,0 +1,129 @@
+    // locally sourced.
+    kmw.addKeyboards({id:'us',name:'English',languages:{id:'eng',name:'English'},
+      filename:'../us-1.0.js'});
+      
+    // Add more keyboards to the language menu, by keyboard name,
+    // keyboard name and language code, or just the ISO 639 language code.  
+    kmw.addKeyboards('french','european2@swe','european2@nor','@heb');
+  
+    // Add a keyboard by language name.  Note that the name must be spelled
+    // correctly, or the keyboard will not be found.  (Using ISO codes is
+    // usually easier.)
+    kmw.addKeyboardsForLanguage('Dzongkha');
+    
+    // Add a fully-specified, locally-sourced, keyboard with custom font  
+    kmw.addKeyboards({id:'lao_2008_basic',name:'Lao Basic',
+      languages:{
+        id:'lao',name:'Lao',region:'Asia',
+        font:{family:'LaoWeb',source:['../font/saysettha_web.ttf','../font/saysettha_web.woff','../font/saysettha_web.eot']}
+        },
+      filename:'../lao_2008_basic.js'
+      });   
+
+    // The following two optional calls should be delayed until language menus are fully loaded:
+    //  (a) a specific mapped input element input is focused, to ensure that the OSK appears
+    //  (b) a specific keyboard is loaded, rather than the keyboard last used.         
+  //window.setTimeout(function(){kmw.setActiveElement('ta1',true);},2500);
+  //window.setTimeout(function(){kmw.setActiveKeyboard('Keyboard_french','fra');},3000);
+  
+    // Note that locally specified keyboards will be listed before keyboards 
+    // requested from the remote server by user interfaces that do not order
+    // keyboards alphabetically by language.
+  }
+  
+    // Script to allow a user to add any keyboard to the keyboard menu 
+  function addKeyboard(n)
+  { 
+    var sKbd, kmw=window['keyman'] ? keyman : tavultesoft.keymanweb;
+    switch(n)
+    {
+      case 1:
+        sKbd=document.getElementById('kbd_id1').value;
+        kmw.addKeyboards(sKbd);
+        break;
+      case 2:
+        sKbd=document.getElementById('kbd_id2').value.toLowerCase();
+        var rx=new RegExp(/^\w{3,3}$\$?/);    
+        if(rx.test(sKbd))
+          kmw.addKeyboards('@'+sKbd);
+        else        
+          alert('An ISO 639 language code must be exactly 3 letters long!');
+        break;
+      case 3:
+        sKbd=document.getElementById('kbd_id3').value;
+        kmw.addKeyboardsForLanguage(sKbd);
+        break;
+    }
+  }
+  
+  // Add keyboard on Enter (as well as pressing button)
+  function clickOnEnter(e,id)
+  {                                       
+    e = e || window.event;
+    if(e.keyCode == 13) addKeyboard(id); 
+  }
+  
+  function removeKeyboard() {
+    var kmw=window['keyman'] ? keyman : tavultesoft.keymanweb;
+    
+    var sKbd = document.getElementById('kbd_id4').value;
+    var result = kmw.removeKeyboards(sKbd);
+    
+    console.log("Keyboard '" + sKbd + "' removal success: " + result);
+  }
+  
+  // Add keyboard on Enter (as well as pressing button)
```

Comment:
copy-paste typo?

Reply:
Yeah, I'm a big fan of CTRL+C CTRL+V at times...